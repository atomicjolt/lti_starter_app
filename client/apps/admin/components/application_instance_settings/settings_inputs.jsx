import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';
import Flatpickr from 'react-flatpickr';
import _ from 'lodash';
import Input from '../common/input';
import Textarea from '../common/textarea';

export default function SettingsInputs(props) {

  const {
    title,
    inputs,
    inputClass,
    textareaClass,
    bodyClass
  } = props;


  return (
    <>
      <div className="aj-settings-title">
        {title}
      </div>
      <div className={`aj-item-padding ${bodyClass}`}>
        {_.map(inputs, (input, index) => {
          if (input.type === 'select') {
            return (
              <div key={index} className={`c-input c-input--container ${inputClass}`}>
                <span>{input.labelText}</span>
                <ReactSelect
                  className={`aj-react-select ${input.className}`}
                  options={input.options}
                  value={input.value}
                  name={input.name}
                  placeholder={input.placeholder}
                  onChange={input.onChange}
                  isClearable={false}
                />
              </div>
            );
          }
          if (input.type === 'textarea') {
            return (
              <Textarea
                key={index}
                className={`c-input aj-input--border ${input.className} ${textareaClass}`}
                labelText={input.labelText}
                textareaProps={input.props}
              />
            );
          }
          if (input.type === 'datepicker') {
            return (
              <div key={index} className={`c-input c-input--container ${inputClass}`}>
                <span>{input.labelText}</span>
                <Flatpickr
                  options={{
                    minDate: input.minDate,
                    maxDate: input.maxDate,
                  }}
                  value={input.value}
                  id={input.id}
                  onChange={(value) => input.onChange(value, input.name)}
                />
              </div>
            );
          }
          return (
            <Input
              key={index}
              className={input.className || `c-input aj-input--border ${inputClass}`}
              labelText={input.labelText}
              inputProps={input.props}
            />
          );
        })}
      </div>
    </>
  );
}

SettingsInputs.propTypes = {
  title: PropTypes.string,
  inputs: PropTypes.array,
  inputClass: PropTypes.string,
  textareaClass: PropTypes.string,
  bodyClass: PropTypes.string,
};
