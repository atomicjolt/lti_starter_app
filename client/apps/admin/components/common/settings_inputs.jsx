import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

export default function SettingsInputs(props) {
  return (
    <div>
      <input
        type="hidden"
        name="oauth_consumer_key"
        value={props.settings.lti_key || ''}
      />
      {
        _.map(props.settings, (value, key) => (
          <input key={key} type="hidden" value={value || ''} name={key} />
        ))
      }
    </div>
  );
}

SettingsInputs.propTypes = {
  settings: PropTypes.shape({
    lti_key: PropTypes.string
  }).isRequired
};
