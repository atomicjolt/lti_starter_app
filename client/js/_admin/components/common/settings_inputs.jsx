import React from 'react';
import _     from 'lodash';

export default function SettingsInputs(props) {
  return (
    <div>
      <input
        type="hidden"
        name="oauth_consumer_key"
        value={props.settings.lti_key}
      />
      {
        _.map(props.settings, (value, key) => (
          <input key={key} type="hidden" value={value} name={key} />
        ))
      }
    </div>
  );
}

SettingsInputs.propTypes = {
  settings: React.PropTypes.shape({
    lti_key: React.PropTypes.string
  }).isRequired
};
