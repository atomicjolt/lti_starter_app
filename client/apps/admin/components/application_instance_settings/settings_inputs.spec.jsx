import React from 'react';
import TestRenderer from 'react-test-renderer';

import { settings_inputs } from './settings_inputs';

describe('settings_inputs', () => {
  it('renders the settings_inputs component', () => {
    const result = TestRenderer.create(
        <settings_inputs />
    );
    expect(result).toMatchSnapshot();
  });
});
