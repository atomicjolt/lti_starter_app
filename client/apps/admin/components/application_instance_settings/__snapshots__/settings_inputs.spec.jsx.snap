// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsInputs renders the SettingsInputs component 1`] = `
Array [
  <div
    className="aj-settings-title"
  />,
  <div
    className="aj-item-padding undefined"
  />,
]
`;
